packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -237,126 +169,170 @@ fn eval_file<'s>(scope: &mut v8::HandleScope<'s>, path: &str) -> Result<Value<'s
+    let mut scope = unsafe { v8::CallbackScope::new(context) };
+    let specifier = specifier.to_rust_string_lossy(&mut scope);
+
+    match ModuleMap::import_module(&mut scope, &specifier) {
+        Ok(module) => Some(module),
+        Err(err) => {
+            tracing::error!("Couldn't import {specifier}, {err}.");
+
+            None
+        }
+    }
 }
 
 impl<'s> Embedded<'s> {
-    fn import(
-        scope: &mut v8::HandleScope<'s>,
-        context: v8::Local<'s, v8::Context>,
-    ) -> Result<Self> {
-        let arrow = eval_file(
-            scope,
-            "./src/worker/runner/javascript/apache-arrow-bundle.js",
-        )?;
+    fn import_common_js_files(scope: &mut v8::HandleScope<'s>) -> Result<Self> {
+        // `hash_stdlib` can't be imported as a module because it needs to be available globally for
+        // behaviors.
         let hash_stdlib = eval_file(scope, "./src/worker/runner/javascript/hash_stdlib.js")?;
-        let hash_util = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/hash_util.js",
-            &[arrow],
-        )?;
-        let batches_prototype = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/batch.js",
-            &[arrow, hash_util],
-        )?;
+        let hash_stdlib_str = new_js_string(scope, "hash_stdlib");
+        scope
+            .get_current_context()
+            .global(scope)
+            .set(scope, hash_stdlib_str.into(), hash_stdlib);
 
-        let ctx_import = import_file(
-            scope,
-            context,
-            "./src/worker/runner/javascript/context.js",
-            &[hash_util],
-        )?;
-        let ctx_import: Array<'_> = ctx_import.try_into().map_err(|err| {
-            Error::FileImport(
-                "./src/worker/runner/javascript/context.js".to_string(),
-                format!("Couldn't get array (of functions) from 'context.js': {err}"),
-            )
-        })?;
-        let experiment_ctx_prototype = ctx_import.get_index(scope, 0).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get experiment_ctx_prototype from index 0 of the context.js return \
-                 values"
-                    .to_string(),
-            )
-        })?;
-        let sim_init_ctx_prototype = ctx_import.get_index(scope, 1).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get sim_init_ctx_prototype from index 1 of the context.js return values"
-                    .to_string(),
-            )
-        })?;
-        let gen_ctx = ctx_import.get_index(scope, 2).ok_or_else(|| {
-            Error::V8(
-                "Couldn't get gen_ctx from index 2 of the context.js return values".to_string(),
-            )
-        })?;
+        let runner = ModuleMap::import_module(scope, "./src/worker/runner/javascript/runner.js")
+            .map_err(|err| {
+                Error::PackageImport(
```

Comment:
Why do we need to wrap the error from `import_module` again with an error called `PackageImport`? This should be handled in the function itself.